### ISSUE TITLE
The `CV_MAKETYPE` macro.

### ISSUE THREAD

**Issue by issuefiler** [UrielCh/opencv4nodejs#54]
# [The `CV_MAKETYPE` macro](https://github.com/opencv/opencv/blob/64aad34cb4abfb6a2603f3f4ecae7f4f0ba1414d/modules/core/include/opencv2/core/hal/interface.h#L69-L86)

 We got constants for depths, constants for numbers of channels, and constants for combinations of both (“_types_”), yet we don’t have any convenient means to _build_ a type in `opencv4nodejs`. I suggest binding [the `CV_MAKETYPE` macro](https://github.com/opencv/opencv/blob/64aad34cb4abfb6a2603f3f4ecae7f4f0ba1414d/modules/core/include/opencv2/core/hal/interface.h#L69-L86).

```C++
#define CV_CN_MAX     512
#define CV_CN_SHIFT   3
#define CV_DEPTH_MAX  (1 << CV_CN_SHIFT)

#define CV_8U   0
#define CV_8S   1
#define CV_16U  2
#define CV_16S  3
#define CV_32S  4
#define CV_32F  5
#define CV_64F  6
#define CV_16F  7

#define CV_MAT_DEPTH_MASK       (CV_DEPTH_MAX - 1)
#define CV_MAT_DEPTH(flags)     ((flags) & CV_MAT_DEPTH_MASK)

#define CV_MAKETYPE(depth,cn) (CV_MAT_DEPTH(depth) + (((cn)-1) << CV_CN_SHIFT))
#define CV_MAKE_TYPE CV_MAKETYPE
```

**Comment by issuefiler** [comment]
The `CV_MAKETYPE` macro is defined as `((n_channels - 1) << 3) | depth` with `depth` being an integer within 0–7 and `n_channels` being an integer within 1–512.

**Comment by UrielCh** [comment]
C macros are not exposed to NodeJS code, so the readability improvement of having a new macro stay limited to the internal C code base.

So...

**Comment by issuefiler** [comment]
@UrielCh

I meant porting the C++ macro into Javascript/Typescript and having it as a utility method. Yes, it wouldn’t be a direct binding.

It was just a suggestion, as it seemed useful to the users. I wouldn’t mind if you want to keep this package limited to direct bindings, though….

**Comment by UrielCh** [comment]
16F is missing, So the issue can be merged with #52

**Comment by UrielCh** [comment]
I have added 16F in the macro as a first move, but since 16F is not supported by CPU, it will be hard to get started with.
